### DISCUSSION TITLE
ToC generator or plugin support for VSCode

### DISCUSSION THREAD

**Discussion by Jeetah** [marp-team/marp#377] [Ideas]
It would be awesome if Marp for VSCode would support adding, generating or even auto-updating a ToC. There exist VSCode plugins for MD - but those do not auto-update very well together with Marp. Perhaps this is possible via Marpit plugins (like adding [[toc]] )? I know there exist some - but it is not easily possible to use those with the VSCode UI.

**Comment by yhatt** [comment]
Probably VS Code extensions for generating ToC will work correctly if implemented marp-team/marp-core#299.

Refer to https://github.com/marp-team/marp-core/issues/136#issuecomment-564388487 and https://www.npmjs.com/search?q=keywords%3Amarkdown-it-plugin%20toc about Marp/Marpit compatible markdown-it plugins. You must use Marp CLI whenever using plugins.
